test/blocks/gist/gist.test.js

```diff
@@ -6,10 +7,46 @@ document.body.innerHTML = await readFile({ path: './mocks/body.html' });
 const { default: init } = await import('../../../libs/blocks/gist/gist.js');
 
 describe('adobetv autoblock', () => {
-  it('creates iframe video block', async () => {
+  let createElementStub;
+  let clock;
+  let originalCreateElement;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    originalCreateElement = document.createElement;
+
+    createElementStub = sinon.stub(document, 'createElement').callsFake((tagName) => {
+      if (tagName !== 'script') {
+        return originalCreateElement.call(document, tagName);
+      }
+      const scriptMock = originalCreateElement.call(document, 'script');
+      Object.defineProperty(scriptMock, 'src', {
+        set(url) {
+          const gistCb = url.match(/callback=([^&]+)/);
+          if (!gistCb) return;
+          const callbackName = gistCb[1];
+          setTimeout(() => {
```

Comment:
Similar comment compared to the Promise one
> Why is this executed within a Promise resolve? should you rather use clock.runAllAsync if you awiat for a cb to execute?

See https://sinonjs.org/releases/latest/fake-timers/